packages/@react-aria/dnd/stories/Reorderable.tsx

```diff
@@ -166,8 +142,17 @@ function ReorderableGrid(props) {
     onDropExit: chain(action('onDropExit'), console.log),
     onDropActivate: chain(action('onDropActivate'), console.log),
     onDrop: async e => {
-      onDrop(e);
-      props.onDrop?.(e);
+      if (e.target.type !== 'root' && e.target.dropPosition !== 'on' && props.onMove) {
+        let keys = [];
+        for (let item of e.items) {
+          if (item.kind === 'text' && item.types.has(dragType)) {
+            let key = JSON.parse(await item.getText(dragType));
```

Comment:
Interesting, do we have to wait for them sequentially? or could we throw them all into a Promise.all and run them in parallel?

```
     let promises = [];
     for (let item of e.items) {
          if (item.kind === 'text' && item.types.has(dragType)) {
            promises.push(item.getText(dragType));
     ...
     let keys = await Promise.all(promises).then(keys => keys.map(key => JSON.parse(key)));
     props.onMove(keys, e.target);
```

Reply:
I don't think this really matters as reading the text is < 1ms. Just a story anyway.